### PR TITLE
Add golink for Q1 DevIntel

### PR DESCRIPTION
**Pull request by ericwindmill**
Adds a go-link for Q1 2024 DevIntel.

## Presubmit checklist

- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

**Review by craiglabenz — APPROVED** [review]

File: firebase.json [hunk 1]
```diff
@@ -321,6 +321,7 @@
       { "source": "/go/detect-memory-leaks", "destination": "https://docs.google.com/document/d/1hvRVpULPnwGjBc5nw52iSaDfB8hjIzbaQuECHmKvDrY/edit", "type": 301 },
       { "source": "/go/developing-plugins", "destination": "/packages-and-plugins/developing-packages#plugin", "type": 301 },
       { "source": "/go/devintel-q3-2023", "destination": "https://docs.google.com/document/d/1VGux5IxePO8lN_7kUuzYJtcJ76fkldr_TJv4R0tciYk", "type": 301 },
+      { "source": "/go/devintel-q1-2024", "destination": "https://docs.google.com/document/d/15W9wsqzFcidXR12DiPH5r5ZDcZw_wDa7Zi9CAAe34Y4", "type": 301 },
```
Comment (parlough) [inline firebase.json:324]: The link doesn't seem to be viewable with just the link.

**Review by parlough — CHANGES_REQUESTED** [review]

File: firebase.json (hunk 1 above)
Comment (ericwindmill) [reply firebase.json]: I'm curious why the link directly above this one works without any additional args? Embedded images? 
 
 Either way, thanks for catching.

Comment (parlough) [reply firebase.json]: I haven't figured out definitively when a resource key is needed or not. I thought it was related to the doc owner's organization configuration, but it seems inconsistent :P

**Review by parlough — APPROVED** [review]
Thanks for updating that!

**Pull request merged**